andela/bp-esa-backend
# #162989999 Replace mock allocation POST endpoints with generated placement

#### What does this PR do?
Replace mock allocation POST endpoint with autogenerated placement data

#### Description of Task to be completed?
- Define `mockPlacement` function to generate unique placement data whenever a request to andela allocations endpoint is made in development environment
- Modify the `fetchNewPlacements` function to filter placements where  `createdAt >= currentDate() - timer_interval`, so as to prevent repeated automations on the same data in development

#### How should this be manually tested?
1. Fetch current branch: `git fetch origin ch-replace-mock-allocations-api-with-placement-seeder-162989999`
2. Checkout to this branch
3. Install necessary dependencies: `npm install`
4. Ensure to set the `ANDELA_API_BASE_URL` variable to `http://localhost:8000/mock-api` in .env file
5. Set development `TIMER_INTERVAL` to `10s`.
6. Start the server: `npm start` and wait for `10s` for the automation to be carried out
7. Observe that new automation data were saved to the database, which means that placement data was generated automatically

#### Any background context you want to provide?
The aim is to simplify the complex  steps being carried out when testing automation endpoints, so as to make it easier for a new developer

#### What are the relevant pivotal tracker stories?
[#162989999](https://www.pivotaltracker.com/story/show/162989999)
